README.md

```diff
@@ -12,16 +12,274 @@
 
 Check out the [ROADMAP](ROADMAP.md) to see what's coming.
 
+## Table of Contents
+
+- [Dependencies](#dependencies)
```

Comment:
since the readme is long, maybe generate a more detailed ToC with https://github.com/thlorenz/doctoc?